### PR TITLE
RDART-1021: Wire up some basic dynamic setting and change notifications

### PR DESCRIPTION
**Pull request by nirinchev**
Fixes https://github.com/realm/realm-dart/issues/1668

Additionally, had to add `Realm.dynamic.create` and `RealmObject.dynamic.set` to be able to test.

TODO:
- [x] Change notifications for collections of objects
- [x] Tests for setting of different property types
- [x] Changelog

### CODE REVIEW

**Comment by nirinchev** [comment]
I don't think this should conflict too much with #1550, but happy to merge the handles PR first.

File: packages/realm_dart/test/dynamic_realm_test.dart [hunk 1]
```diff
@@ -222,6 +226,95 @@ void main() {
...
+      actual.dynamic.set('intProp', 42);
+      actual.dynamic.set('nullableIntProp', -42);
+
+      actual.dynamic.set('decimalProp', Decimal128.fromDouble(500));
+      actual.dynamic.set('nullableDecimalProp', Decimal128.infinity);
+
+      actual.dynamic.set('realmValueProp', RealmValue.from([true, 5]));
+    });
+
+    expect(actual.dynamic.get('stringProp'), isNot(expected.stringProp));
```
Comment (elle-j) [inline packages/realm_dart/test/dynamic_realm_test.dart:261]: Would it not be preferable to test against the exact updated value instead of `isNot` here?

**Review by elle-j — APPROVED** [review]
LGTM 🙂

File: packages/realm_dart/test/dynamic_realm_test.dart (hunk 1 above)
Comment (nirinchev) [reply packages/realm_dart/test/dynamic_realm_test.dart:261]: Sure, we could also do that - I was mostly lazy here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
 ## vNext (TBD)
 
 ### Enhancements
-* None
+* Added support for creating and storing a RealmObject using the `Realm.dynamic` API: `realm.dynamic.create("Person", primaryKey: 123)`. (PR [#1669](https://github.com/realm/realm-dart/pull/1669))
+* Added support for setting properties on a RealmObject using the dynamic API: `obj.dynamic.set("name", "Peter")`. (PR [#1669](https://github.com/realm/realm-dart/pull/1669))
 
 ### Fixed
-* None
+* Listening for `.changes` on a dynamic object (obtained via the `realm.dynamic` API) no longer throws. (Issue [#1668](https://github.com/realm/realm-dart/issues/1668))
```
[the remark this answers is not in the document]
Comment (nirinchev) [reply CHANGELOG.md:8]: I guess you can argue both ways - it was reported as a bug, but technically we didn't intend it to work, so maybe it's an enhancement.

[the remark this answers is not in the document]
Comment (elle-j) [reply CHANGELOG.md:8]: Gotcha, I think I would see it more of an enhancement then.

File: packages/realm_dart/test/dynamic_realm_test.dart (hunk 1 above)
Comment (elle-j) [reply packages/realm_dart/test/dynamic_realm_test.dart:261]: Then we can catch potential incorrect updates 👍

**Pull request merged**